pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -103,6 +103,55 @@ func MonEndpoints(mons map[string]*cephclient.MonInfo, requireMsgr2 bool) []stri
 	return endpoints
 }
 
+// updateNetNamespaceFilePath modify the netNamespaceFilePath for all cluster IDs.
+// If holderEnabled is set to true. Otherwise, removes the netNamespaceFilePath value
+// for all the clusterIDs.
+func updateNetNamespaceFilePath(clusterNamespace string, cc csiClusterConfig) {
+	var (
+		cephFSNetNamespaceFilePath string
+		rbdNetNamespaceFilePath    string
+		nfsNetNamespaceFilePath    string
+	)
+
+	if IsHolderEnabled() {
+		for _, centry := range cc {
+			if centry.Namespace == clusterNamespace {
+				if centry.CephFS.NetNamespaceFilePath != "" {
+					cephFSNetNamespaceFilePath = centry.CephFS.NetNamespaceFilePath
+				}
+				if centry.RBD.NetNamespaceFilePath != "" {
+					rbdNetNamespaceFilePath = centry.RBD.NetNamespaceFilePath
+				}
+				if centry.NFS.NetNamespaceFilePath != "" {
+					nfsNetNamespaceFilePath = centry.NFS.NetNamespaceFilePath
+				}
+			}
```

Comment:
This might work but i would suggest another approach, you have the clusterNamespace with you , you can generate the namespacePath using `GenerateNetNamespaceFilePath` and substitute the values for all the entries where the clusterNamespace and x.Namespace is matching? WDYT?

Reply:
`GenerateNetNamespaceFilePath` requires other params as well. We can do it, but will need more change.
Since, we have already called `GenerateNetNamespaceFilePath` before and set the values we just need to copy it to other clusterIDs of same clusterNamespace.
https://github.com/rook/rook/blob/1a50482868ace760693f9f9880e1d463714ec7ff/pkg/operator/ceph/csi/spec.go#L933-L938

Reply:
Yes i agree i was looking for more non error prone approach, if someone had updated the CM with some dummy value, we are going to use it for all clusterID's which is not correct

Reply:
👍🏻